Admin/AclAdminExtension.php

```diff
@@ -42,16 +43,22 @@ class AclAdminExtension extends AdminExtension
     protected $roleHierarchy;
 
     /**
-     * @param SecurityContextInterface $securityContext
-     * @param Connection               $databaseConnection
-     * @param array                    $roleHierarchy
+     * @param SecurityContextInterface|TokenStorageInterface $tokenStorage
+     * @param Connection                                     $databaseConnection
+     * @param array                                          $roleHierarchy
      */
     public function __construct(
-        SecurityContextInterface $securityContext,
+        $tokenStorage,
         Connection $databaseConnection,
         array $roleHierarchy = array()
     ) {
-        $this->securityContext = $securityContext;
+        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
+            throw new \InvalidArgumentException('$securityContext must be an instanse of SecurityContextInterface or of TokenInterface '
+                    . 'but [' .  get_class($tokenStorage) . '] given.');
```

Comment:
> Can you remove this part of the exception message?

@meyerbaptiste  which code should I use to replace it?

Previously you wrote that:

>Then test inside the constructor if $securityContext is an instance of TokenStorageInterface or SecurityContextInterface. If not, throw an InvalidArgumentException

and now my exception is quite informative, isn't it?
If not pleace write me replacement code)